crates/bic/src/main.rs

```diff
@@ -306,10 +303,7 @@ fn do_diff(
     let newer_contents = fs::read(newer).context("read new file")?;
 
     let (mut patch_r, mut patch_w) = pipe::pipe();
-    let diff_params = DiffParams {
-        sort_partitions: *sort_partitions,
-        scan_chunk_size: *scan_chunk_size,
-    };
+    let diff_params = DiffParams::new(*sort_partitions, *scan_chunk_size).unwrap();
```

Comment:
I tend to dislike positional parameters but I can't think of a good solution that also does validation. Makes me wonder if validation should happen when *creating DiffParams* or when trying to use it anywhere.

There's also the builder pattern but... lots of boilerplate or extra proc-macro dependencies aren't great either.

Not sure what to do here~

Reply:
Yeah I had the same thoughts, I was even about to use NonZeroUsize but that's a lot of hassle for little gain. Doing a builder pattern here would be overkill, too, IMHO. And with rust-analyzer it actually now insert the param names as hints. Let's review this when we add a third param?

Reply:
Fair enough